core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -198,10 +198,19 @@ public StructuredRecord joinOn(String stageName, StructuredRecord record) {
     List<Schema.Field> fields = new ArrayList<>();
     Schema schema = record.getSchema();
 
+
     List<String> joinKeys = perStageJoinKeys.get(stageName);
     int i = 1;
     for (String joinKey : joinKeys) {
-      Schema.Field joinField = Schema.Field.of(String.valueOf(i++), schema.getField(joinKey).getSchema());
+      Schema.Field field = schema.getField(joinKey);
+      if (field == null) {
+        collector.addFailure(
+          String.format("Join key field '%s' is not present in input stage of '%s'.", joinKey, stageName), null)
```

Comment:
Join key field '[field name]' does not exist in schema from '[stage name]'

Reply:
fixed.